apps/snoopdb/postgres/snoopUtils.py

```diff
@@ -242,86 +226,78 @@ def format_uri_parts_for_namespace_finalize(uri_parts):
     uri_second_half =['{name}','finalize']
     return uri_first_half + uri_second_half
 
-def find_operation_id(openapi_spec, event):
-  method=assign_verb_to_method(event)
-  url = urlparse(event['requestURI'])
-  # 1) Cached seen before results
-  # Is the URL in the hit_cache?
-  if url.path in openapi_spec['hit_cache']:
-    # Is the method for this url cached?
-    if method in openapi_spec['hit_cache'][url.path].keys():
-      # Useful when url + method is already hit multiple times in an audit log
-      return openapi_spec['hit_cache'][url.path][method]
-    # part of the url of the http/api request
-  uri_parts = url.path.strip('/').split('/')
-  # IF we git a proxy component, the rest of this is just parameters and we don't "count" them
+def format_uri_parts(path):
+  uri_parts = path.strip('/').split('/')
   if 'proxy' in uri_parts:
     uri_parts = format_uri_parts_for_proxy(uri_parts)
-  # namespace status and finalize endpoints are also a lil different
-  if is_namespace_status(uri_parts):
+  elif is_namespace_status(uri_parts):
       uri_parts = format_uri_parts_for_namespace_status(uri_parts)
-  if is_namespace_finalize(uri_parts):
+  elif is_namespace_finalize(uri_parts):
       uri_parts = format_uri_parts_for_namespace_finalize(uri_parts)
+  return uri_parts
+
+def is_ignored_endpoint(uri_parts):
+    if any(part in uri_parts for part in IGNORED_PATHS):
+        return True
+    if uri_parts == ['openapi','v2']:
+        return True
+    return False
+
+# given an open api spec and audit event, returns operation id and an error.
+# If the opID can be found in the spec,
+# then we return it with a nil error.
+# Otherwise, we return a nilID and a given error message.
+# we add both op id and error to our events,
+# so that we can parse events by error in snoopdb
+def find_operation_id(openapi_spec, event):
+  method=assign_verb_to_method(event['verb'], event['requestURI'])
+  if method is None:
+      return None, "Could not assign a method from the event verb. Check the event.verb."
+  url = urlparse(event['requestURI'])
+  if url.path in openapi_spec['hit_cache']:
+    if method in openapi_spec['hit_cache'][url.path].keys():
```

Comment:
From our review: not sure the Pythonic way to do it but merge into a single statement
```python
if thing == True and
  otherthing == True:
```